### PR TITLE
fix: cleaning up registerRegistry after instantiate

### PR DESCRIPTION
**Pull request by k-j-kim**
- Fixes https://github.com/systemjs/systemjs/issues/2124
- Deletes entries off `registerRegistry` after instantiating
- Introduces `nameRegistry` to keep track of resolved module names

### CODE REVIEW

File: dist/extras/named-register.js [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
   function setRegisterRegistry(systemInstance) {
     systemInstance.registerRegistry = Object.create(null);
+    systemInstance.nameRegistry = [];
```
Comment (guybedford) [inline dist/extras/named-register.js]: It would be faster to use an object here, since arrays are O(n) lookup.

**Comment by guybedford** [comment]
Another alternative that might work is simply setting `registerRegistry[name] = null`, where we can then check `registerRegistry.hasOwnProperty(name)` for the resolver.

That might save some code and minor perf.

**Comment by k-j-kim** [comment]
good call; I've removed `nameRegistry` in favor of setting the registry entry to null.

**Comment by k-j-kim** [comment]
thank you for the prompt feedbacks!

**Comment by k-j-kim** [comment]
> I know of several systemjs users who rely on named register resolve to continue working after the first use of it, and this would break that.

@joeldenning can you clarify how this would break the named register resolve? Since we aren't deleting the entry and only setting it as null, `id in this.registerRegistry` will still return the id:
```js
  systemJSPrototype.resolve = function (id, parentURL) {
    try {
      // Prefer import map (or other existing) resolution over the registerRegistry
      return resolve.call(this, id, parentURL);
    } catch (err) {
      if (id in this.registerRegistry) {
        return id;
      }
      throw err;
    }
  };
```
> Additionally, could you please remove the changes to dist in this PR?

I've done that in the latest commit. Thanks!

**Comment by joeldenning** [comment]
published in https://github.com/systemjs/systemjs/releases/tag/6.2.4

**Comment by k-j-kim** [comment]
fantastic, thanks @joeldenning !

**Pull request merged**